### ISSUE TITLE
Remove sortish_sampler from Seq2SeqTrainingArgument and the docs

### ISSUE THREAD

**Issue by eyalmazuz** [huggingface/transformers#34986]
### System Info

Not Needed

### Who can help?

@muellerzr 
@SunMarc 
@stevhliu 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

In  [Seq2SeqTrainingArguments](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.Seq2SeqTrainingArguments) There's a parameter called sortish_sampler

* sortish_sampler (bool, optional, defaults to False) — Whether to use a sortish sampler or not. Only possible if the underlying datasets are Seq2SeqDataset for now but will become generally available in the near future.
It sorts the inputs according to lengths in order to minimize the padding size, with a bit of randomness for the training set.

This argument exists in the Seq2SeqTrainer 

https://github.com/huggingface/transformers/blob/9d0ad5074eb4214efa8063de21ac5253470f4e08/src/transformers/training_args_seq2seq.py#L39 

But it does nothing aside from taking space
In the Trainer class it points to nothing, never referenced, never used, and in the trainer_utils there isn't even a class called SortishSampler

Personally I didn't knew it, I used this argument for a long while now thinking it does something in my training scripts
but only upon entering the source code and seeing it doesn't exists, I started a deep dive into old issues and pull requests.

Then I found #9574 which is a (almost) 4 years old PR
in which it was renamed and technically got deprecated.

I suggest to remove the argument completely from the documentation and trainer code to remove confusion

As personally I though that ``group_by_length`` is meant mostly for auto-regressive models which only takes a single sequence as their input, while ``sortish_sampler`` is used for seq2seq models and takes into account both sequences

So I decided to use the latter rather than the former for the past 3 weeks thinking I'll get some memory savings in my scripts.

In the end, I think this argument should officially get deprecated and removed from both the code and the docs and it may cause confusion for others beside me.

### Expected behavior

Less confusion, better model training :)

**Comment by Rocketknight1** [comment]
cc @muellerzr @SunMarc

**Comment by eyalmazuz** [comment]
@muellerzr 
@SunMarc

**Comment by SunMarc** [comment]
Thanks for the report @eyalmazuz ! Feel free to deprecate completely the arg as it is already doing nothing at the moment as you said. We do have some legacy code related to Seq2seq Trainer (examples/lecacy/seq2seq) where this arg is still used but since it is legacy, we won't touch the code there.
